posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+                    sh '''
+                         mvn -U package
+                         docker login <remote-docker-image-repository-url> -u "${USERNAME}" -p “${PASSWORD}”
+                         docker build -t liberty-$<code identifier>:$<docker image version> .
+                         #eg: docker build -t liberty-app:v1.0 .
+                         docker tag liberty-$<code identifier>:$<docker image version> <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker tag liberty-app:v1.0 docker.io/someid/liberty-app:v1.0
+                         docker push <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker push docker.io/someid/liberty-app:v1.0  
+                       '''   
+                  }
+               }
+            }
+          }
+
+
+Following are the parameter that is used in this example code
+
+* Git token: Personal access token generated from github.
+* remote-docker-image-repository-url : Docker hub repository URL.
+* USERNAME/PASSWWORD: username and password to connect to docker registry.
+
+image::/img/blog/jenkins-cred-binding-and-corresponding-param.png[Pipeline credential binding and corresponding param,width=30%,align="center"]
+
+* code identifier: This is optional, a unique docker image identifier
+* docker image version: docker image version number, a unique identifier   
+
+*Security scan* can be done along with Maven build command (CLI commands way of application scanning) or can do it in a separate pipeline stage. 
+
+For static code analysis, SonarQube community edition can be used. Install SonarQube server by either using file start-up type from cli by downloading the package in link:https://www.sonarqube.org/success-download-community-edition/[SonarQube server installation package] or use docker way as explained in link:https://docs.sonarqube.org/latest/setup/get-started-2-minutes/[SonarQube server installation steps]. SonarQube jenkins client setup details are given at link:https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-jenkins/[SonarQube client for jenkins]. 
+ 
+    CLI command example, integrated with Maven command: 
+    $ mvn package verify sonar:sonar -Dsonar.projectKey=sampleapp -Dsonar.host.url=http://localhost:9000 -Dsonar.login=7b13f240123456780479b13d09ba8
+   
+For Docker image scan, you can use link:https://hub.docker.com/r/aquasec/trivy/[Docker image scan with trivy]. This scan provides the vulnerability details of open source jars that are used in the application. Sample commands are given below
+
+       sh '''
+             docker run aquasec/trivy image $docker_login/sampleapp:v1.0
+             #eg: docker run aquasec/trivy image docker.io/someid/liberty-app:v1.0
+       '''
+
```

Comment:
Remove this detail. It's too much detail that they can find in the respective docs.

Reply:
This is actually very less details in the complete setup, there are a lot more to explain in general, how to interpret scan results regarding code vulnerabilities and coverage percentage, how to push selected scan results to server, how to fail the build job in case of not satisfying coverage percentage, how to develop quality gate setup, where can we see the test case execution details in case of failures, why is the coverage percentage zero in case of failures etc etc. its a long list and nothing is explained in this section. Actually none of these things will be available on official documents directly)  . I have removed the line "*Security scan* can be done along with Maven build command (CLI commands way of application scanning) or can do it in a separate pipeline stage. ", other lines are very important for the clarity. Steps followed are like this. First, explained the SonarQube server setup, on both VM and containerized environment. Then the client setup, an example is given after that following the pattern in other sections. We can discuss this further.